codalab-service.py

```diff
@@ -0,0 +1,387 @@
+
+        start_cmd.add_argument('--build-locally', '-b', action='store_true', help='If specified build VERSION using local code.', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--test-build', '-t', action='store_true', help='If specified run tests on the build.', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--user-compose-file', type=str, help='If specified path to a user-defined Docker compose file that overwrites the defaults', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--start-worker', '-w', action='store_true', help='If specified start a CodaLab worker on this machine.', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--initial-config', '-i', action='store_true', help='If specified, save the initial configuration of the instance (defaults to true if the service home or the database mounts are ephemeral)', default=argparse.SUPPRESS)
+
+        #  USER CREDENTIALS
+
+        start_cmd.add_argument('--mysql-root-pwd', type=str, help='Root password for the database', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--mysql-user', type=str, help='MYSQL username for the Codalab MYSQL client', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--mysql-pwd', type=str, help='MYSQL password for the Codalab MYSQL client', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--root-user', type=str, help='Codalab username for the Codalab admin user', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--root-pwd', type=str, help='Codalab password for the Codalab admin user', default=argparse.SUPPRESS)
+
+        #  HOST FILESYSTEM MOUNTS
+
+        start_cmd.add_argument('--uid', type=str, help='Linux UID that owns the files created by Codalab. default=(ID of the user running this script)', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--service-home', type=str, help='Path on the host machine to store home directory of the Codalab server (by default nothing is stored', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--mysql-mount', type=str, help='Path on the host machine to store mysql data files, by default the database is ephemeral', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--worker-dir', type=str, help='Path on the host machine to store worker data files, by default these are ephemeral', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--bundle-store', type=str, help='Path on the host machine to store bundle data files, by default these are ephemeral', default=[], dest='bundle_stores', action='append')
+
+        start_cmd.add_argument('--worker-docker-network-name', type=str, help='Name of the docker network that includes the worker and runs', default=argparse.SUPPRESS)
+
+        #  HOST PORT MOUNTS
+
+        start_cmd.add_argument('--http-port', type=str, help='HTTP port for the server to listen on', default=argparse.SUPPRESS)
```

Comment:
if ssl is setup, then this is https? (document)

Reply:
As things stand, HTTPS always listens on port 443 and this actually only changes the HTTP port regardless of whether you're using SSL or not. 

After your comment, I think it makes more sense to have it modify HTTPS if SSL enabled, HTTP otherwise.